### DISCUSSION TITLE
multiple USB tethered RNDIS (LTE) modems possible? (Hostless Modems)

### DISCUSSION THREAD

**Discussion by nmz787** [SmoothWAN/SmoothWAN#15] [General & Help]
rural situation here. Two 4G LTE modems on different cell towers, with different weather dependent issues. MultiWAN on FreshTomato isn't working as expected, and their GUI doesn't support multiple USB tethered LTE modems, so I attempted to run two routers with freshtomato running on each, each having one of the LTE modems attached, then connected a LAN port to the WAN port of the other and enabling multiWAN there. It's terrible. Possibly because of multiple subnets and me being terrible at linux networking.

Anyway, can I run two USB tethered LTE modems on a SmoothWAN raspi 4?

**Comment by TalalMash** [comment]
Hello!

Yes in fact that is the recommended way to use LTE modems with SmoothWAN: RNDIS or Hi-link (Huawei) mode.
QMI and NCM are discouraged since each dongle requires different settings, and difficult to configure SMS and USSD support.
Simply plug in the modems and follow the basic instructions in the README page, they will both appear as USB ethernet adapters. 

Note: You may have to plug in the modem to a PC first, change the IP address if both modems use the same IP, this step is only needed if you need to access modem's configuration page for SMS/USSD. 
Example with a Huawei: 
![image](https://user-images.githubusercontent.com/96490382/158057711-7695f6a6-b1b4-4035-ad96-bc465fd2eac5.png)

**Reply by evanrodgers** [reply]
Hello Talal, thanks again for your help in the other thread.

I have a Quectel EM12-G in a USB3.0 enclosure (that works in Windows) but it's not appearing in SmoothWAN UI at all. It looks like the modem is detected (see the image attached), but it's not showing up anywhere. And idea what to do?

Thanks!
![image](https://user-images.githubusercontent.com/36175609/162208621-e3be414b-a726-4a94-b2a5-96bd213d06ae.png)

EDIT: Also, perhaps relatedly, I do not see the Multi-WAN USB option in the network tab? I do have a different modem (an MR1100) attached via USB and it shows up normally in the Speedify UI, but the EM12-G is missing.

![image](https://user-images.githubusercontent.com/36175609/162210560-191c3931-ed58-43b9-9c51-a60c4191f4d2.png)

**Reply by TalalMash** [reply]
Hello!

No problem! Regarding that modem, its seems to be NCM/QMI modem-only which is not supported as plug and play, its impossible to predict the settings for these types. You would have to add it manually in "Interfaces" section like the following example:

1. In Network->Interfaces, click `Add new interface...`
![image](https://user-images.githubusercontent.com/96490382/162211059-3c4f232c-7335-412b-9776-bec75289c894.png)
2. Quacetel seems to ship many variants with different defaults, I'm going to assume it supports QMI:
![image](https://user-images.githubusercontent.com/96490382/162212434-78fdcddd-baf6-4658-bba8-a0b0823cf479.png)
3. You may have to use different `Modem device` field like /dev/ttyUSB0 or 1 etc.. as well as other ISP related settings.
![image](https://user-images.githubusercontent.com/96490382/162212591-db7b69df-fbc2-4c23-95ba-00bb460abc3a.png)
4. Set gateway metric, the lower the number the high priority in usage when Speedify is down
![image](https://user-images.githubusercontent.com/96490382/162212925-640914cf-42c2-4e83-9209-ce4ef3ef3e39.png)
5. Set to WAN in firewall tab, done!
![image](https://user-images.githubusercontent.com/96490382/162213029-8ab73e6c-74ce-42d2-8d92-63141d988a11.png)

Multi-WAN USB menu is not available on x86 as its a generic image, it's not possible for me to predict or test every machine.
I could add device support based on users feedback and a well lit picture of the ports.

**Reply by evanrodgers** [reply]
Okay, amazing. That makes sense re: Multi-WAN USB and generic x86, no worries there.

What mode should these modems be in for them to work plug-and-play with SmoothWAN? I'm sure I can switch the mode, just need to know what to search.

**Reply by TalalMash** [reply]
> Okay, amazing. That makes sense re: Multi-WAN USB and generic x86, no worries there.
> 
> What mode should these modems be in for them to work plug-and-play with SmoothWAN? I'm sure I can switch the mode, just need to know what to search.


Unfortunately these modems are meant to be shipped with a host, there are other portable models with a firmware that supports hostless mode with RNDIS. This modem does support Host-mode RNDIS mode but it requires configuration with AT commands which is not plug and play.
If you have managed to get it working post the settings set here and I'll add them in 0.99.6 in QMI mode, it seems to be the most reliable mode compared to MBIM and NCM.
There seems (forum) to be many issues when using multiple of these on a Windows host due to fixed plug and play settings.

**Reply by umekzg** [reply]
#Set Connection Modes for EM12-G
AT+QCFG="usbnet",0 - QMI/PPP/Default
AT+QCFG="usbnet",1 - ECM
AT+QCFG="usbnet",2 - MBIM
